src/schemes/fluid/surface_tension.jl

```diff
@@ -174,8 +174,9 @@ end
                                        neighbor_system::AbstractFluidSystem,
                                        particle, neighbor, pos_diff, distance,
                                        rho_a, rho_b, grad_kernel)
-    # No cohesion with oneself
-    distance < sqrt(eps()) && return zero(pos_diff)
+    # No cohesion with oneself.
+    # Handle numerical precision issues (see also https://github.com/trixi-framework/TrixiParticles.jl/pull/913).
```

Comment:
```suggestion
    # No cohesion with oneself. See `src/general/smoothing_kernels.jl` for more details.
```
Same for all other changes below.